tracing-subscriber/src/fmt/format.rs

```diff
@@ -280,8 +309,54 @@ where
     }
 }
 
-// === impl FormatFields ===
+#[cfg(feature = "json")]
+impl<N, T> FormatEvent<N> for Format<Json, T>
+where
+    N: for<'writer> FormatFields<'writer>,
+    T: FormatTime,
+{
+    fn format_event(
+        &self,
+        ctx: &span::Context<'_, N>,
+        writer: &mut dyn fmt::Write,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let mut timestamp = String::new();
+        self.timer.format_time(&mut timestamp)?;
+
+        #[cfg(feature = "tracing-log")]
+        let normalized_meta = event.normalized_metadata();
+        #[cfg(feature = "tracing-log")]
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
+        #[cfg(not(feature = "tracing-log"))]
+        let meta = event.metadata();
+
+        let mut visit = || {
+            let mut serializer = Serializer::new(WriteAdaptor::new(writer));
+            let mut serializer = serializer.serialize_map(None)?;
+
+            serializer.serialize_entry("timestamp", &timestamp)?;
+            serializer.serialize_entry("level", &meta.level().as_serde())?;
+
+            ctx.with_current(|(_, span)| serializer.serialize_entry("span", span.name()))
+                .unwrap_or(Ok(()))?;
 
+            if self.display_target {
+                serializer.serialize_entry("target", meta.target())?;
+            }
+
+            let mut visitor = SerdeMapVisitor::new(serializer);
```

Comment:
I'm on the fence about this, but let me know what you think: should the fields be serialized as a nested object or list on the serialized event? 

For example, do you think it's better to serialize events like this:
```json
{
    "timestamp": "...",
    "level": "INFO",
    "target": "tracing_subscriber::fmt::format::test",
    "fields": {
        "message": "my_message",
        "some_other_field": "some value",
     }
}
```
or like this:
```json
{
    "timestamp": "...",
    "level": "INFO",
    "target": "tracing_subscriber::fmt::format::test",
    "message": "my_message",
    "some_other_field": "some value",
}
```

Reply:
When you think one of the prime use cases here, using it in some log aggregator, it would be kind of nice to get the custom fields mapped to its own object. So I'd say the first option would be pretty nice.

Reply:
This escalated a bit :(

Some changes needed for `tracing-core` to get this done in an easy way.

https://github.com/pimeys/tracing/commit/b143824b9e1390f79527e7e680a00f324ea9046c